File: Client/src/types/Category.ts
export interface Category {
    categoryId: number;
    name: string;
    slug: string;
}

export interface CategoryForm {
    id: string | number;
    name?: string;
    slug?: string;
    parentId?: number | null;
    parentName?: string;
    status?: string;
    createDate?: string;
    createdBy?: string;
    updateDate?: string;
    updatedBy?: string;
}

File: Domain/Entities/Category.cs
using DTech.Domain.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace DTech.Domain.Entities;

public partial class Category
{
    [Key]
    public int CategoryId { get; set; }

    [Required(ErrorMessage = "Please enter name of category")]
    public string? Name { get; set; }

    [Display(Name = "Parent")]
    public int? ParentId { get; set; }

    public string? Slug { get; set; }

    [Display(Name = "Created By")]
    public string? CreatedBy { get; set; }

    [Display(Name = "Create Date")]
    public DateTime? CreateDate { get; set; }

    [Display(Name = "Updated By")]
    public string? UpdatedBy { get; set; }

    [Display(Name = "Update Date")]
    public DateTime? UpdateDate { get; set; }

    public StatusEnums Status { get; set; } = StatusEnums.Available;

    public virtual ICollection<Category> InverseParent { get; set; } = new List<Category>();

    public virtual Category? Parent { get; set; }

    public virtual ICollection<Product> Products { get; set; } = new List<Product>();
}
